providers.py

```diff
@@ -177,7 +177,7 @@ def msg_push(self, body, repo):
             last_commit_url = body['commits'][-1]['url']
             commit_messages = [
                 dict(msg=c['message'][:80].split('\n')[0], hash=c['id'][:8],
-                     url=c['url']) for c in body['commits']
+                     url=c['url'][:-28]) for c in body['commits']
```

Comment:
Can you elaborate on this change? E.g:

1. Why is this intended to remove a part of the url?
2. Is this only for repos on gitlab.com? What if we're using a self hosted instance (with a different domain), will this still work?
3. Is the intent to remove query-params? If so, maybe a regex or [`urllib.parse`][1] is a better fit?

[1]: https://docs.python.org/3/library/urllib.parse.html#module-urllib.parse